### DISCUSSION TITLE
User workspace data access

### DISCUSSION THREAD

**Discussion by lauriseitsonen** [EOEPCA/eoepca#113] [Data Access]
Hi,

I would like to get WMS access to staged-out products from user processes using the data access component deployed for the user but I'm struggling with its configuration. It does not seem to work out of the box - do you have a working configuration? 

BR, Lauri

**Comment by rconway** [comment]
Hi @lauriseitsonen 
My expectation would be that, with the `auto_create_product_types` setting of the data-access in the workspace templates - ref. https://github.com/EOEPCA/deployment-guide/blob/main/deploy/eoepca/workspace-templates/template-hr-data-access.yaml#L106 - then I would expect the products to be available through eoxserver.
@constantinius Is that your expectation?

Do you see the products in the Resource Catalogue of the user's workspace?

**Reply by lauriseitsonen** [reply]
Hi,

With the default settings I could see the product in user's Resource Catalogue as a Collection item within metadata:main collection, with only Links property. 
![kuva](https://github.com/EOEPCA/eoepca/assets/973769/e09673b9-5b33-4215-877f-4df682c4ecde)

Harvester showed an error like
`{"event": "FileNotFoundError('ws-eric/processing-results/2bd14d98-2eee-11ef-bf20-ba6f9d0d1421/2bd14d98-2eee-11ef-bf20-ba6f9d0d1421/collection.json/catalog.json')...`

I then modified the cookie cutter template for the process (service.py line 282) and set it to register the original catalog.json (from variable s3_path). After that user's Resource Catalogue showed a new top level Collection for the product and the product as a Record in that collection (as well as in the metadata:main collection)
![kuva](https://github.com/EOEPCA/eoepca/assets/973769/4f24d494-abbb-4c5b-bd32-8d421b9d0688)

The WMS link opens the getCapabilities xml and I can make a getMap request for the layer. The getMap request for layers=ID__outlines shows the bounding box as expexted but request for layers=ID__outlined or layers=ID return a blank image. The pod log for renderer shows the query and http 200 response. So the product seems to be partly in the eox view server but something is missing.

**Comment by rconway** [comment]
The strange `xxx/collection.json/catalog.json` path in the original registration is indeed strange. Can you share your fix.

Nevertheless, it seems you got the STAC registered so would expect it to be rendered via eox view server. What is the format of the output?

**Reply by lauriseitsonen** [reply]
The fix was to replace 
`"url": collection.get_self_href()`
with
`"url": s3_path`
on line
https://github.com/EOEPCA/eoepca-proc-service-template/blob/a270701ff98b1e294821f3e255256137d833d0f1/%7B%7Bcookiecutter.service_name%7D%7D/service.py#L282
(in my case the fix is local to a single process at the moment as I modified the file on the Zoo NFS share)

The WMS requests produce image/png output. A sample query that would produce a transparent png:
`SERVICE=WMS&VERSION=1.3.0&REQUEST=GetMap&FORMAT=image%2Fpng&TRANSPARENT=true&LAYERS=test_tif_GdalWarp_1718869407710275610__outlined&CRS=EPSG%3A3857&STYLES=magenta&WIDTH=800&HEIGHT=800&BBOX=1500000%2C6200000%2C1700000%2C6400000`

**Reply by lauriseitsonen** [reply]
It seems that the registration process does not create a default browse type for products and that's why their content is not rendered. After I manually created a browse type for a product WMS requests for the product rendered as expected. I used the following command for a product that contained a single band GeoTiff file as asset file_1 to get greyscale output:
`python3 manage.py browsetype create cdb82802e36f41d3a2ac7dcda319eaff --grey file_1 --grey-range 0 4000`

**Reply by rconway** [reply]
@lauriseitsonen Thanks for the info

**Comment by rconway** [comment]
Hi @constantinius 
Is this a bug - or maybe there is some additional configuration?
What do you think?
